### ISSUE TITLE
Move to Java 8

### ISSUE THREAD

**Issue by AlexLandau** [ggp-org/ggp-base#126]
This has come up before and been blocked on the fact that Tiltyard is a major consumer of GGP-Base as a library, and it uses Google App Engine, which still doesn't have Java 8 support.

However, the Google App Engine team has announced that they will be supporting Java 8 in the next few months: https://cloudplatform.googleblog.com/2017/03/your-favorite-languages-now-on-Google-App-Engine.html

Given that, it's now worth tracking this in its own ticket.

(Note that in the meantime, if you have your own checkout of GGP-Base and want to use Java 8, you should feel free to modify the sourceCompatibility and targetCompatibility in build.gradle to 1.8 and regenerate your IDE files with e.g. `./gradlew eclipse` or `./gradlew idea`. This should enable you to use Java 8 language features.)

**Comment by teohhanhui** [comment]
Looks like this is outdated as Java 7 is no longer supported on Google App Engine?

https://cloud.google.com/appengine/docs/flexible/java/configuring-your-app-with-app-yaml

**Comment by jk2500** [comment]
Are there any plans for moving to Java 8?

**Comment by AlexLandau** [comment]
@jk2500 Neither Sam nor I have worked on this repo for some time. Per the last paragraph in the first post, you should be able to edit the Gradle configuration to change the language version yourself and the existing code should be compatible with 8 or (possibly) with later versions.

Some features of more recent Java releases (sealed types and more convenient syntax around `instanceof`) would work nicely with the `Gdl*` types, but I don't sense enough external interest in GGP to motivate me to revisit this project.
